library/src/main/java/org/apache/apex/malhar/lib/wal/FSWindowDataManager.java

```diff
@@ -484,6 +484,10 @@ private Object retrieve(FSWindowReplayWAL wal, long windowId) throws IOException
 
       if (wal.retrievedWindow == null) {
         wal.retrievedWindow = readNext(reader);
+        if(wal.retrievedWindow == null &&
+           reader.getCurrentPointer().compareTo(wal.walEndPointerAfterRecovery) < 0) {
+          continue;
+        }
         Preconditions.checkNotNull(wal.retrievedWindow);
```

Comment:
There is an assumption that `wal.retrievedWindow  != null` that does not seems to be valid.